commit ad9675e47b495285e3be62a0d9c87edffd0ae8c3
Author: Tomi Valkeinen <tomi.valkeinen@ti.com>
Date:   Thu Jan 3 13:59:51 2019 +0200

    drm/bridge: tc358767: fix initial DP0/1_SRCCTRL value
    
    [ Upstream commit 9a63bd6fe1b5590ffa42ae2ed22ee21363293e31 ]
    
    Initially DP0_SRCCTRL is set to a static value which includes
    DP0_SRCCTRL_LANES_2 and DP0_SRCCTRL_BW27, even when only 1 lane of
    1.62Gbps speed is used. DP1_SRCCTRL is configured to a magic number.
    
    This patch changes the configuration as follows:
    
    Configure DP0_SRCCTRL by using tc_srcctrl() which provides the correct
    value.
    
    DP1_SRCCTRL needs two bits to be set to the same value as DP0_SRCCTRL:
    SSCG and BW27. All other bits can be zero.
    
    Signed-off-by: Tomi Valkeinen <tomi.valkeinen@ti.com>
    Reviewed-by: Andrzej Hajda <a.hajda@samsung.com>
    Signed-off-by: Andrzej Hajda <a.hajda@samsung.com>
    Link: https://patchwork.freedesktop.org/patch/msgid/20190103115954.12785-5-tomi.valkeinen@ti.com
    Signed-off-by: Sasha Levin <sashal@kernel.org>

diff --git a/drivers/gpu/drm/bridge/tc358767.c b/drivers/gpu/drm/bridge/tc358767.c
index fee53422c31f..ab299f4debfa 100644
--- a/drivers/gpu/drm/bridge/tc358767.c
+++ b/drivers/gpu/drm/bridge/tc358767.c
@@ -836,12 +836,11 @@ static int tc_main_link_setup(struct tc_data *tc)
 	if (!tc->mode)
 		return -EINVAL;
 
-	/* from excel file - DP0_SrcCtrl */
-	tc_write(DP0_SRCCTRL, DP0_SRCCTRL_SCRMBLDIS | DP0_SRCCTRL_EN810B |
-		 DP0_SRCCTRL_LANESKEW | DP0_SRCCTRL_LANES_2 |
-		 DP0_SRCCTRL_BW27 | DP0_SRCCTRL_AUTOCORRECT);
-	/* from excel file - DP1_SrcCtrl */
-	tc_write(DP1_SRCCTRL, 0x00003083);
+	tc_write(DP0_SRCCTRL, tc_srcctrl(tc));
+	/* SSCG and BW27 on DP1 must be set to the same as on DP0 */
+	tc_write(DP1_SRCCTRL,
+		 (tc->link.spread ? DP0_SRCCTRL_SSCG : 0) |
+		 ((tc->link.base.rate != 162000) ? DP0_SRCCTRL_BW27 : 0));
 
 	rate = clk_get_rate(tc->refclk);
 	switch (rate) {
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1385_ad9675e47b495285e3be62a0d9c87edffd0ae8c3.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1385_ad9675e47b495285e3be62a0d9c87edffd0ae8c3.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
